### ISSUE TITLE
Consider deferring trie lookup for characters below U+0300 in the NFC iterator case

### ISSUE THREAD

**Issue by hsivonen** [unicode-org/icu4x#2385]
#2378 optimizes away trie lookups when NFC passthrough succeeds. However, that's only with contiguous buffers. With the iterator formulation, characters that are equal or above `decomposition_passthrough_bound` but below `composition_passthrough_bound` could potentially avoid a trie lookup, but currently a trie lookup is made anyway.

Figure out if there's a reasonably-manageable way to defer the trie lookup in the iterator case.

**Comment by sffc** [comment]
@hsivonen What would be an appropriate milestone for this issue?

**Comment by hsivonen** [comment]
This isn't urgent, so it's kinda weird to assign a milestone. Since this doesn't need to be in 1.0 and not necessarily in next, either. I put 1.2 as the milestone.

**Comment by hsivonen** [comment]
This is unlikely to be worthwhile after #6925, since #6925 makes the perf delta between this and not having this smaller.

I'm a bit shy to WONTFIX this outright in case someone else has great ideas of how to do this without over-complicating code.
